Sources/MapboxNavigation/CarPlayManager.swift

```diff
@@ -657,6 +657,10 @@ extension CarPlayManager: CPMapTemplateDelegate {
 
 @available(iOS 12.0, *)
 extension CarPlayManager: CarPlayNavigationDelegate {
+    public func carPlayNavigationViewController(_ carPlayNavigationViewController: CarPlayNavigationViewController, shouldPresentArrivalUIFor waypoint: Waypoint) -> Bool {
```

Comment:
This time I'm getting different crash:
[Example_2021-05-24-190445_MacBook-Pro.txt](https://github.com/mapbox/mapbox-navigation-ios/files/6533618/Example_2021-05-24-190445_MacBook-Pro.txt)

It happens when arriving to final destination, but when screen, which was shown for intermediate waypoint on CarPlay wasn't dismissed.

Reply:
Thanks for the suggestion, I'll work on the logic of the template presenting.

Reply:
This time it's better, thanks. Getting another crash though :) To reproduce this run navigation session as in similar scenario, after arriving to the final destination dismiss presented view and start navigation once again. You should get crash similar to this one:

[Example_2021-05-25-141400_MacBook-Pro.txt](https://github.com/mapbox/mapbox-navigation-ios/files/6538791/Example_2021-05-25-141400_MacBook-Pro.txt)

Please also do several runs with usual/unusual flows, maybe there are several more issues/crashes we haven't noticed yet related to presentation of such views.

Reply:
Thanks for the suggestion, I'll concentrate on testing today to discover the problems.

Reply:
Thanks for the suggestion. After the tests, sometimes the map camera may not be at current location when canceling the navigation or ending the navigation on iOS or CarPlay.